src/libtiled/wangset.cpp

```diff
@@ -134,6 +187,71 @@ void WangId::flipVertically()
     rotate(2);
 }
 
+WangIdVariationIterator::WangIdVariationIterator(int edgeColors, int cornerColors, WangId wangId)
```

Comment:
So the solution I've put together implements a proper iterator to replace to old messy one. It works for the end case by using the biggest wangId for given input + 1 to the underlying id. This works great for all sets, other than those 15x15, where this value will be zero, and immediately fail. As a work around, everywhere this iterator is used must be wrapped with an if testing if it is the 15x15 case, and if so takes on slightly different behavior.